pkg/router/controller/unique_host.go

```diff
@@ -131,6 +131,24 @@ func (p *UniqueHost) HandleRoute(eventType watch.EventType, route *routeapi.Rout
 				}
 			}
 			if !added {
+				// Clean out any old form of this route
+				next := []*routeapi.Route{}
+				for i := range old {
+					if old[i].Name != route.Name {
```

Comment:
Use routeNameKey(old[i]) != routeNameKey(route) instead of old[i].Name != route.Name